ext/standard/http_fopen_wrapper.c

```diff
@@ -203,6 +203,13 @@ static php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 
 	if (context && (tmpzval = php_stream_context_get_option(context, wrapper->wops->label, "timeout")) != NULL) {
 		double d = zval_get_double(tmpzval);
+
+		if (d > (double) PHP_TIMEOUT_ULL_MAX / 1000000.0) {
+			php_stream_wrapper_log_error(wrapper, options, "timeout must be lower than " ZEND_ULONG_FMT, (zend_ulong)((double) PHP_TIMEOUT_ULL_MAX / 1000000.0));
+			zend_string_release(transport_string);
+			php_url_free(resource);
+			return NULL;
+		}
 #ifndef PHP_WIN32
 		timeout.tv_sec = (time_t) d;
 		timeout.tv_usec = (size_t) ((d - timeout.tv_sec) * 1000000);
```

Comment:
The `size_t` cast looks wrong. `timeout.tv_usec` is an `suseconds_t` which is [according to POSIX](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/sys_types.h.html):

> The type suseconds_t shall be a signed integer type capable of storing values at least in the range [-1, 1000000].

Probably okay to leave this for stable branches, but I suggest to change to `(suseconds_t)` in master.

The `(long)` casts on Windows are, unfortunately, correct. From WinSock2.h:
````c
struct timeval {
        long    tv_sec;         /* seconds */
        long    tv_usec;        /* and microseconds */
};
````

Reply:
Yes I would definitively keep this existing code untouched in stable branch.